src/libraries/System.Text.Json/src/System/Text/Json/Serialization/StreamReadBufferState.cs

```diff
@@ -51,26 +51,25 @@ public ReadBufferState(int initialBufferSize)
 
         public readonly bool IsFinalBlock => _isFinalBlock;
 
-        public readonly ReadOnlySpan<byte> Bytes => _buffer.AsSpan(_offset, _count);
+        public readonly ReadOnlySequence<byte> Bytes => new(_buffer.AsMemory(_offset, _count));
```

Comment:
I take it we're switching to sequence-based Utf8JsonReaders for streams with this change? Given we're close to RC, I'm slightly concerned that any issues with that implementation would surface as regressions in the stream-based APIs, is there any chance we could keep this implementation in span mode? One way to do that is delegate Utf8JsonReader construction to a IReadBufferState method.

Reply:
We look at whether the ReadOnlySequence is a single segment and use the Span overload in that case. Stream will always be a single segment so there won't be any regressions there.

Reply:
Sure, but even so the instance is constructed using the multi-segnent modality so different code paths are being entered.

Reply:
It is still constructing the `Utf8JsonReader` with the `ReadOnlySpan` overload
<img width="1705" height="608" alt="image" src="https://github.com/user-attachments/assets/9070a78d-b50e-434d-ab15-58e408332249" />

The only difference is that we first create a `ReadOnlySequence` https://source.dot.net/#System.Memory/System/Buffers/ReadOnlySequence.cs,113

We can change `IReadBufferState` to have a construction method like you mentioned, but it doesn't change what is happening.